src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -154,18 +154,6 @@ public bool ContainsAssemblyOrModuleOrDynamic(ISymbol symbol, bool primary)
                     return new CompilationTracker(newProject, newState);
                 }
 
-                var declarationOnlyCompilation = state.DeclarationOnlyCompilation;
```

Comment:
This might be your failing test problem: before, if the various compilations were GC'ed, we'd still have the DeclarationOnlyCompilation which would hold onto the previously ran generators. Now we're going to just hit the "new CompilationTracker from scratch" path below.